### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by forabi**
Closes hollowverse/hollowverse#472

### CODE REVIEW

File: src/typings/p-memoize.d.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    /**
+     * Use a different cache storage.
+     * Must implement the following methods:
+     * `.has(key)`, `.get(key)`, `.set(key, value)`, `.delete(key)`, and optionally `.clear()`
+     * You could for example use a `WeakMap` instead or `quick-lru` for a LRU cache.
+     *
+     * @default new Map()
+     */
+    cahce?: Map | WeakMap;
```
Comment (wholesomedev) [inline src/typings/p-memoize.d.ts]: Typo? :)

Comment (forabi) [reply src/typings/p-memoize.d.ts]: Oops!

Comment (forabi) [reply src/typings/p-memoize.d.ts]: BTW, we are using `p-memoize` in many repos so I submitted the type definitions to DefinitelyTyped (https://github.com/DefinitelyTyped/DefinitelyTyped/pull/25773)

**Review by wholesomedev — COMMENTED** [review]
LGTM, but the logs are not being parsed correctly for some reason. I filed a related issue https://github.com/hollowverse/hollowverse/issues/478

**Pull request merged**